This isn't working; the check fails with:

```
 In file included from /__w/imu_tools/imu_tools/src/imu_tools/imu_complementary_filter/src/complementary_filter_ros.cpp:34:
/__w/imu_tools/imu_tools/src/imu_tools/imu_complementary_filter/include/imu_complementary_filter/complementary_filter_ros.h:37:10: fatal error: sensor_msgs/msg/magnetic_field.hpp: No such file or directory
   37 | #include <sensor_msgs/msg/magnetic_field.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

According to ament/ament_cmake#572, there is a bit more work to do than just replacing `ament_target_dependencies` -> `target_link_libraries`. You also have to replace 

* `geometry_msgs` -> `${geometry_msgs_TARGETS}`
* `message_filters` -> `message_filters::message_filters`
* ...

But which one's which? According to the ament PR, there should be a very helpful deprecation notice that you can just copy&paste, but somehow I cannot get it in the GitHub actions environment. I'm not sure if the deprecation notice is even already there in the current rolling Docker images. Since Kilted isn't released yet, there are also no Docker images for Kilted yet, which makes it even harder to test.

--- a/imu_complementary_filter/CMakeLists.txt
+++ b/imu_complementary_filter/CMakeLists.txt
@@ -25,13 +25,14 @@ target_include_directories(complementary_filter PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
   $<INSTALL_INTERFACE:include>)
 target_link_libraries(complementary_filter PUBLIC
-  geometry_msgs
-  message_filters
-  rclcpp
-  sensor_msgs
-  std_msgs
-  tf2
-  tf2_ros
+  ${geometry_msgs_TARGETS}
+  ${sensor_msgs_TARGETS}
+  ${std_msgs_TARGETS}
+  message_filters::message_filters
+  rclcpp::rclcpp
+  sensor_msgs::sensor_msgs_library
+  tf2::tf2
+  tf2_ros::tf2_ros
 )
 # Causes the visibility macros to use dllexport rather than dllimport,
 # which is appropriate when building the dll but not consuming it.

--- a/imu_filter_madgwick/CMakeLists.txt
+++ b/imu_filter_madgwick/CMakeLists.txt
@@ -23,13 +23,15 @@ target_include_directories(imu_filter_madgwick PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
   $<INSTALL_INTERFACE:include>)
 target_link_libraries(imu_filter_madgwick PUBLIC
-  geometry_msgs
-  rclcpp
-  rclcpp_components
-  sensor_msgs
-  std_msgs
-  tf2_geometry_msgs
-  tf2_ros
+  ${geometry_msgs_TARGETS}
+  ${sensor_msgs_TARGETS}
+  ${std_msgs_TARGETS}
+  rclcpp::rclcpp
+  rclcpp_components::component
+  rclcpp_components::component_manager
+  sensor_msgs::sensor_msgs_library
+  tf2_geometry_msgs::tf2_geometry_msgs
+  tf2_ros::tf2_ros
 )
 rclcpp_components_register_nodes(imu_filter_madgwick "ImuFilterMadgwickRos")
 target_compile_definitions(imu_filter_madgwick

--- a/rviz_imu_plugin/CMakeLists.txt
+++ b/rviz_imu_plugin/CMakeLists.txt
@@ -61,7 +61,8 @@ add_library(${PROJECT_NAME} SHARED ${SRC_FILES})
 ## the ``find_package(Qt4 ...)`` line above, or by the
 ## ``set(QT_LIBRARIES Qt5::Widgets)``, and with whatever libraries
 ## catkin has included.
-target_link_libraries(${PROJECT_NAME} PUBLIC ${dependencies} rviz_ogre_vendor::OgreMain rviz_ogre_vendor::OgreOverlay Qt5::Widgets)
+
+target_link_libraries(${PROJECT_NAME} PUBLIC rviz_ogre_vendor::OgreMain rviz_ogre_vendor::OgreOverlay Qt5::Widgets ${sensor_msgs_TARGETS} message_filters::message_filters rviz_common::rviz_common rviz_ogre_vendor::OgreMain rviz_rendering::rviz_rendering sensor_msgs::sensor_msgs_library tf2::tf2 tf2_ros::tf2_ros)
 
 # prevent pluginlib from using boost
 target_compile_definitions(${PROJECT_NAME} PUBLIC "PLUGINLIB__DISABLE_BOOST_FUNCTIONS")

